Resources/Prototypes/XenoArch/Effects/utility_effects.yml

```diff
@@ -0,0 +1,153 @@
+﻿# Utility effects permanently modify the entity in some way when triggered, and they generally make it 'useful' for some purpose,
+# like turning the artifact into a tool, or gun, or whatever.
+
+- type: artifactEffect
+  id: EffectStorage
+  targetDepth: 2
+  effectHint: artifact-effect-hint-storage
+  restrictStructures: true # it doesnt necessarily have to be restricted from structures, but i think it'll be better that way
+  permanentComponents:
+  - type: Storage
+    capacity: 50
+  - type: UserInterface
+    interfaces:
+    - key: enum.StorageUiKey.Key
+      type: StorageBoundUserInterface
+  - type: ContainerContainer
+    containers:
+      storagebase: !type:Container
+        ents: []
```

Comment:
What happens when you revisit one of these nodes? (Nothing good. Make it check if it can actually apply the components before it does anything.)

Reply:
pretty sure permanent components dont retrigger (though i will do that anyway)